frameit/lib/frameit/editor.rb

```diff
@@ -277,24 +271,28 @@ def put_title_into_background(background, stack_title)
       sum_width = title.width
       sum_width += keyword.width + keyword_padding if keyword
 
-      # Resize the 2 labels if necessary
-      smaller = 1.0 # default
-      ratio = (sum_width + (keyword_padding + horizontal_frame_padding) * 2) / image.width.to_f
-      if ratio > 1.0
-        # too large - resizing now
-        smaller = (1.0 / ratio)
-
-        UI.verbose("Text for image #{self.screenshot.path} is quite long, reducing font size by #{(ratio - 1.0).round(2)}")
+      title_below_image = fetch_config['title_below_image']
 
-        title.resize("#{(smaller * title.width).round}x")
-        keyword.resize("#{(smaller * keyword.width).round}x") if keyword
-        sum_width *= smaller
+      # Resize the 2 labels if they exceed the available space either horizontally or vertically:
+      image_scale_factor = 1.0 # default
+      ratio_horizontal = sum_width / (image.width.to_f - horizontal_frame_padding * 2) # The fraction of the text images compared to the left and right padding
+      ratio_vertical = title.height.to_f / actual_font_size # The fraction of the actual height of the images compared to the available space
+      if ratio_horizontal > 1.0 || ratio_vertical > 1.0
+        # If either is too large, resize with the maximum ratio:
+        image_scale_factor = (1.0 / [ratio_horizontal, ratio_vertical].max)
+
+        UI.verbose("Text for image #{self.screenshot.path} is quite long, reducing font size by #{(100 * (1.0 - image_scale_factor)).round(1)}%")
+        # When the height is reduced more than the width, show a hint to use 'font_scale_factor':
+        UI.message("Hint: Your text height is reduced to fit, probably you use multi-line text. Please consider to use parameter 'font_scale_factor' in your Framefile.json with a value larger than #{fetch_config['font_scale_factor'] || 0.1}. This will create more space #{title_below_image ? 'below' : 'above'} the device frame.") if ratio_vertical > ratio_horizontal
```

Comment:
Do you mean the hint is annoying, or the fact that the screenshot gets a reduced font size?

Reply:
Or both 🤒

Reply:
Scaling the text makes sense; I want the text to fit within the image. I think if the user is explicitly putting in a line break either they either:

 - Will be using all-multiline labels, at which point the message isn't really helping; or
 - Will be using mixed single- and multi-line labels, which will cause the height of the text to change between screenshots

In the latter case, it could be argued it's useful to be notified of the scaling, but I think this is an edge case and mentioning `font_scale_factor` (this is the only documented place it gets mentioned?) would only "fix" the issue for the multi-line labels, and would break the single-line labels.

Reply:
Do I understand correctly that you're OK by only removing the hint?

Reply:
Yes, I think removing the hint would be good :)

Reply:
OK, I'll make another commit.